fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java

```diff
@@ -1560,14 +1560,20 @@ private void getExecHostPortForFragmentIDAndBucketSeq(TScanRangeLocations seqLoc
                     break;
                 }
             }
-
-            buckendIdToBucketCountMap.put(buckendId, buckendIdToBucketCountMap.get(buckendId) + 1);
             Reference<Long> backendIdRef = new Reference<Long>();
             TNetworkAddress execHostPort = SimpleScheduler.getHost(buckendId, seqLocation.locations, idToBackend, backendIdRef);
             if (execHostPort == null) {
                 throw new UserException("there is no scanNode Backend");
             }
-
+            if (backendIdRef.getRef() != buckendId) {
```

Comment:
This logic isn't very simple. would better add comments here for each if-else branch meaning

Reply:
Thanks for your code review. The comments has been added.